hf_vision_model_vertex_ai/locust/local_load_test.py

```diff
@@ -0,0 +1,133 @@
+"""
+User client for Locust for testing Vertex AI endpoints.
+Code modified from https://bit.ly/3bUIyfl
+"""
+
+import json
+import logging
+import os
+import time
+from functools import partial
+from typing import Dict, List
+
+import google.auth
+from google.auth.transport.requests import AuthorizedSession
+from locust import User, constant, events
+
+PROJECT_ID = os.environ["GCP_PROJECT"]
+REGION = os.environ["REGION"]
+ENDPOINT_ID = os.environ["ENDPOINT_ID"]
+JSON_PAYLOAD = os.environ["JSON_PAYLOAD"]
+
+
+class AIPPClient(object):
+    """
+    A convenience wrapper around AI Platform Prediction REST API.
+    """
+
+    def __init__(self, service_endpoint):
+        logging.info(
+            f"Setting the AI Platform Prediction service endpoint: {service_endpoint}"
+        )
+        credentials, _ = google.auth.default()
```

Comment:
maybe worth leaving a note how this works like we need to run `gcloud auth application-default login` beforehand or so
: https://google-auth.readthedocs.io/en/master/reference/google.auth.html